src/chains/admin.py

```diff
@@ -38,6 +41,15 @@ class ChainAdmin(admin.ModelAdmin[Chain]):
     )
     inlines = [FeatureInline, GasPriceInline, WalletInline]
 
+    def get_readonly_fields(
+        self, request: HttpRequest, obj: Optional[Chain] = None
+    ) -> List[str]:
+        if request.user.groups.filter(name="support"):  # type: ignore
```

Comment:
What about having a new table for the warning messages? ChainId + Message.

The endpoint can return the message from this table. The permission then can be done on the table level. Maybe it's easier and simpler this way wdyt?

Reply:
I would prefer not if it's possible to do it in another way, I would prefer to have a normalized database if possible

Reply:
I was thinking more about this field and in a way it seems that it's not really part of the chain metadata so that's why I thought about a separate table. Even though it's possible in a different way the Chain model itself already has quite some fields. But just an idea (nothing that would block the feature).

Reply:
I was thinking in having a separated table as well, it could be nice to keep an history of the changes as well, but I initially discarded it because it seems a bit overkill for the feature. 

About the custom permission... it was my initial approach as well, but in this case we need to still grant edit permission over all Chain fields to superusers, and only over `warning` field for a selected group of users. I thinking in creating an additional permission, something like `can_change_all_chain_fields` and when a user doesn't have it, show just the `warning` as editable. Do you see a simpler way here?

Reply:
I think I'm missing something, but why not using `if request.user.has_perm("app.permission_name")`?

Reply:
Maybe it's me the one missing something, but we need to avoid _support_ group from modifying any field except for `warning` right? So it's kind of _negative permission_. I mean, all the other fields are the ones to restrict to some users, right?

Reply:
I see it more like:
- By default users don't have any permission unless you make them superusers or give them permissions to every table individually.
- `support` group will grant the `edit chains`  and `change_only_warning` to all his members.
- To edit the chains table users will need the `edit chains` permission
- People with `change_only_warning` permission will be able to modify only `warning` field, so `if request.user.has_perm(`change_only_warning`):` every field but warning will be read_only

Reply:
@Uxio0 thanks for your feedback and guidance on this 🙂. I was doing some tests and it seems that when creating a new permission (`change_only_warning` in this case), that permission is automatically inherited by superuser(s). So, in this case, that converts all fields in `read_only` for the superusers also (except `warning` of course) 🤔 I see rwo options here:
- Rely on the group
- Have an additional check to verify the user is not a `superadmin` before marking the fields as read-only.

```python
if (
        request.user.has_perm("change_only_warning")
        and not request.user.is_superuser
    ):
```

I think the second option is better, what do you think?

Reply:
I implemented the second option, of course if you see any concerns with it let me know 🙂